twine/commands/upload.py

```diff
@@ -60,6 +60,11 @@ def upload(upload_settings: settings.Settings, dists: List[str]) -> None:
     repository_url = cast(str, upload_settings.repository_config["repository"])
 
     print(f"Uploading distributions to {repository_url}")
+    if upload_settings.verbose:
+        for filename in uploads:
+            file_size = utils.get_file_size(filename)
+            print(f"  {filename} ({file_size})")
+        print("\n")
```

Comment:
Breaking that logic apart might be nice @VikramJayanthi17 

I'm starting to question the value in printing the list in advance because it's not as if it's part of a prompt. So when we talk about --verbose we're not talking about increasing noise. We're trying to relatively increase volume of output while keeping a high amount of signal.

Reply:
I think there's value in "logging some indication of what's about to happen right before it happens", in the event that what happens is an error. I don't think it matters too much where this happens, but it should probably be as close to the action as we can make it.

Here, if we want to include what signatures will be uploaded along with the file (which I think would be good), it sounds like we should break the loop apart like @VikramJayanthi17 suggests and log a message right before `repository.upload(package)` that includes everything that's about to happen.

Reply:
I like @VikramJayanthi17's proposal as well. I was wondering something similar, but noted that the signing is currently skipped if the package is already uploaded:

https://github.com/pypa/twine/blob/bffcba666875fce1a8e01e3cb991c840e18dd0f9/twine/commands/upload.py#L83-L85

Not sure what the implications of that are.

Reply:
I attempted to address this in the latest [commit](https://github.com/pypa/twine/pull/652/commits/bb971a8deaf5f72dfe97ada0af4fad447d9b1dbf) and outlined my approach in this [comment](https://github.com/pypa/twine/pull/652#issuecomment-642926319).